rscripts/metafeature_nucleotides.R

```diff
@@ -0,0 +1,628 @@
+    
+    # > gene_seq
+    # # A tibble: 1 x 2
+    # Gene  sequence                                            
+    # <chr> <chr>                                               
+    # 1 MAT   AAAAAGAAAACAAAATAAAAATGGCCACATGATTTTTGTTTTCTTTTATTTT
+    
+    seq <- unlist(strsplit(gene_seq$sequence, ''))[left:right]
+    # For gene 'MAT'
+    # seq 
+    # [1] "A" "T" "G" "G" "C" "C" "A" "C" "A" "T" "G" "A"
+    
+    nt_tibble <- tibble(Gene = gene, Pos = 1:length(seq), Nucleotide = seq)
+    
+  }
+  
+  nt_tibble <- purrr::map_df(.x = names, .f = ConvertSequenceToNt, GeneSequenceTibble)
+  
+  # >  head(nt_tibble)
+  # # A tibble: 6 x 3
+  # Gene    Pos Nucleotide
+  # <chr> <int> <chr>     
+  #   1 MAT       1 A         
+  # 2 MAT       2 T         
+  # 3 MAT       3 G         
+  # 4 MAT       4 G         
+  # 5 MAT       5 C         
+  # 6 MAT       6 C 
+  
+}
+
+# nt_tibble <- CreateNtAnnotation(genome, names, gff_df)
+
+# nt_tibble
+# # A tibble: 27 x 3
+# Gene    Pos Nucleotide
+# <chr> <int> <chr>     
+#   1 MAT       1 A         
+# 2 MAT       2 T         
+# 3 MAT       3 G         
+# 4 MAT       4 G         
+# 5 MAT       5 C         
+# 6 MAT       6 C         
+# 7 MAT       7 A         
+# 8 MAT       8 C         
+# 9 MAT       9 A         
+# 10 MAT      10 T         
+# # ... with 17 more rows
+
+
+# Get the location of reads on the transcript for an individual gene
+
+GetReadPositions <- function(gene, dataset, hd_file, asite_displacement_length ,reads_asitepos, left, right){
+  
+  reads_pos_length <- GetGeneDatamatrix(gene, dataset, hd_file) # Get the matrix of read counts
+  
+  reads_asitepos <- CalcAsiteFixed(reads_pos_length, min_read_length, asite_displacement_length)
+  
+  subset_gff_df_by_gene <- dplyr::filter(.data = gff_df, seqnames == gene) 
+  # where gene = YAL003W
+  
+  left <- as.numeric(dplyr::filter(.data = subset_gff_df_by_gene, type == "CDS") %>%  select(start))
+  # 251
+  
+  right <- as.numeric(dplyr::filter(.data = subset_gff_df_by_gene, type == "CDS") %>%  select(end))
+  # 871
+  
+  cds <- reads_asitepos[left:right]
+  # num [1:621] 811 460 2978 429 251 ...
+  
+}
+
+# get the positions of reads for multiple genes and produce a tibble listing the gene, position and count. 
+
+GetAllPosCounts <- function(gene_names, dataset, hd_file, min_read_length){
+  
+  gene_names <- rhdf5::h5ls(hd_file, recursive = 1)$name
+  
+  GetAllPosCounts1Gene <- function(gene, dataset, hd_file, min_read_length, asite_displacement_length){
+    
+    subset_gff_df_by_gene <- dplyr::filter(.data = gff_df, seqnames == gene) 
+    
+    left <- as.numeric(dplyr::filter(.data = subset_gff_df_by_gene, type == "CDS") %>%  select(start))
+    
+    right <- as.numeric(dplyr::filter(.data = subset_gff_df_by_gene, type == "CDS") %>%  select(end))
+    
+    asite_displacement_length <- ReadAsiteDisplacementLengthFromFile(here::here("data", 
```

Comment:
as the asite_displacement_length is sourced for yeast, should this be a parameter set outside of the functions?